specifyweb/frontend/js_src/lib/components/Preferences/BasePreferences.tsx

```diff
@@ -455,7 +459,7 @@ const fetchResourceData = async (
 const fetchDefaultResourceData = async (
   fetchUrl: string,
   defaultResourceName: string
-): Promise<ResourceWithData> =>
+): Promise<Partial<ResourceWithData>> =>
```

Comment:
not exactly a bug, but the presence of `any` masked an invalid type here (return type is `Partial<ResourceWithData>` not `ResourceWithData`